plugins/plugin-core-impl/src/connection.ts

```diff
@@ -32,18 +32,25 @@ export async function connect (
     try {
       if (socket === undefined || isClosed(await socket)) {
         socket = new Promise<WebSocket>(resolve => {
-          const ws = new WebSocket(`ws://${clientUrl}/${clientId}`)
+          const ws = new WebSocket(`ws://${clientUrl}/${token}`)
           ws.onopen = () => {
             resolve(ws)
-
-            // We need to reply requests in case they are missed.
-            processor.onOpen()
           }
-          ws.onerror = () => {
+          ws.onerror = (reason) => {
+            // Reject all pending operations.
+            processor.reject(reason)
+
             // Force reconnect in 1 second.
-            setTimeout(() => { doConnect() }, 1000) // eslint-disable-line
+            setTimeout(async () => await doConnect(), 1000) // eslint-disable-line @typescript-eslint/no-misused-promises
+          }
+          ws.onmessage = (evt) => {
+            const resp = readResponse(evt.data as string)
+            if (resp.id !== undefined) {
+              processor.process(resp)
+            } else if (resp.result !== undefined) {
+              handler(resp.result as Tx)
+            }
           }
-          ws.onmessage = evt => { processor.process(readResponse(evt.data as string)) }
         })
       }
     } catch (err) {
```

Comment:
There is nothing within `catch` block. Is it intentionally?

Reply:
Done.